clientv3/client.go

```diff
@@ -157,16 +158,22 @@ func (c *Client) Dial(endpoint string) (*grpc.ClientConn, error) {
 		grpc.WithBlock(),
 		grpc.WithTimeout(c.cfg.DialTimeout),
 	}
-	if c.creds != nil {
-		opts = append(opts, grpc.WithTransportCredentials(*c.creds))
-	} else {
-		opts = append(opts, grpc.WithInsecure())
-	}
 
 	proto := "tcp"
-	if url, uerr := url.Parse(endpoint); uerr == nil && url.Scheme == "unix" {
-		proto = "unix"
-		// strip unix:// prefix so certs work
+	if url, uerr := url.Parse(endpoint); uerr == nil {
+		switch url.Scheme {
+		case "unix":
+			proto = "unix"
+		case "http":
+			c.creds = nil
+		case "https":
```

Comment:
turns out --insecure-transport=false is broken
should it be defaulting to false (instead of true) anyway?